### ISSUE TITLE
bug: The undefined error seems to cause subsequent pings to fail

### ISSUE THREAD

**Issue by alex-r89** [PassTheMayo/minecraft-server-util#53]
**Describe the bug:**
I thought I would capture this in a separate issue, because the impact seems to be different, but feel free to close this one as it may be cause by the same bug I raised in [#52 ](https://github.com/PassTheMayo/minecraft-server-util/issues/52).

When pinging an array of servers, if the `undefined` error is thrown, often all subsequent pings will just fail with the same `undefined` error. The image below explains this.

**Code:**
Occurs when pinging an array of servers, which could be done in a number of ways. My code for doing this is:

```js
    let completedPings = 0

    await new Promise((resolve) => {
      for (var i = 0; i < servers.length; i++) {
        const server = servers[i]

        util
          .status(`${server.ip}`, { port: server.port, timeout: 3000 })
          .then((response) => {
            console.log(`[🏓 Success pinging ${server.ip}]`)
            // Do something with the response
          })
          .catch((error) => {
            console.log(`[👎 Error pinging ${server.ip}]`, error)
            // Do something with the error
          })
          .finally(() => {
            completedPings++
            if (completedPings >= servers.length) resolve()
          })
      }
    })

// Continue on with other stuff in the ocde
```

**Expected behavior:**
An `undefined` error being thrown should not cause all subsequent pings to fail with the `undefined` error.

**Additional context**
The image below explains this. As you can see, I ping a local IP which seems to consistently throw the `undefined` error. If this does not cause the execution to hang (which issue [#52 ](https://github.com/PassTheMayo/minecraft-server-util/issues/52) captures), then there is a chance all subsequent pings will also fail with `undefined`.

<img width="677" alt="Screenshot 2021-08-15 at 19 02 15" src="https://user-images.githubusercontent.com/7352884/129541886-1d4625b4-8f38-40a2-9200-cc648bd575a6.png">

**Comment by PassTheMayo** [comment]
Just glancing over the code, it appears that your code execution is not asynchronous. The fast and repetitive call to `status()` might be causing a network error which could lead to the `undefined` error as a result. I'll do some testing tomorrow on this error, but in the mean time, can you confirm that this issue wasn't fixed in `3.6.0` (which was released recently which might fix this)?

**Comment by alex-r89** [comment]
Hey,

Thanks for the reply. No, this issue wasn't fixed in `3.6.0`. 

Interesting that you say the fast and repetitive call to `status()` may be causing the `undefined` error. Is that likely? I'm not too sure how the networking side of it works tbh.

The problem with that is, If i use this code asynchronously, naturally it's a lot, lot slower.

In my test I am calling `status()` on 50 servers, which took ~2 seconds with the code above. If I do this asynchronously, the 50 calls to `status()` takes ~25 seconds.

Scaling this up to 5,000 servers would mean my code above takes ~3 minutes, however asynchronously this would take ~41 minutes - which wouldn't work for my use case because I ping the servers every 5 minutes.

I know that's not really an issue for you or this package, but it sort of makes it unusable when pinging an array of servers.

**Comment by PassTheMayo** [comment]
This seems to have been fixed or have been caused by networking issues on the client side. If anybody has this issue again, I will open this back up.
